### ISSUE TITLE
program output not captured when program does not exist or didn't run "succesfully"

### ISSUE THREAD

**Issue by grasmanek94** [eidheim/tiny-process-library#23]
Tested on: Windows 7
Description: Cannot get process output when process cannot be created (or process didn't run "succesfully")
Code:

		Process process(options.Combine(), "", [](const char *bytes, size_t n) {
			std::cout << "xxyy" << std::string(bytes, n) << "yyxx" << std::endl;
		}, [](const char *bytes, size_t n) {
			std::cout << "xxyy" << std::string(bytes, n) << "yyxx" << std::endl;
		});

Identified cause in source code (process_win.cpp:106):

	  ....
	  BOOL bSuccess = CreateProcess(nullptr, process_command.empty()?nullptr:&process_command[0], nullptr, nullptr, TRUE, 0,
	                                nullptr, path.empty()?nullptr:path.c_str(), &startup_info, &process_info);

	  if(!bSuccess) {
	    CloseHandle(process_info.hProcess);
	    CloseHandle(process_info.hThread);
	    return 0; // here it stops and doesn't proceed to fire the lambda's to capture output
	  }
	  else {
	    CloseHandle(process_info.hThread);
	  }
	
	  if(stdin_fd) *stdin_fd=stdin_wr_p.detach();
	  if(stdout_fd) *stdout_fd=stdout_rd_p.detach();
	  if(stderr_fd) *stderr_fd=stderr_rd_p.detach();
	  ....

Suggested fix:

Fire the lambda's with captured output either way (even if bSuccess == false). It's the callers responsibility to check the exit code anyway, and a separate "binary doesn't exist" code can be emitted.

**Comment by eidheim** [comment]
Thank you, but are you sure there are process output when the process cannot be created? Or if there should be process output if the process cannot be created? Unless I'm mistaken, the current behaviour is correct, but feel free argue against me on this:)

**Comment by eidheim** [comment]
Oh, Unix-like systems seems to output wrong commands when using `sh -c`. This seems to be a Windows issue. Feel free to look around for a solution. The problem is that bSuccess is true (if on MSYS2) but no output is sent through the pipe it seems.

**Comment by eidheim** [comment]
If the process is created through `sh -c`, `sh` will output an error message if the process cannot be created due to for instance missing/incorrect executable. However, if `sh -c` is not used, there will be no output to stdout or stderr at all, and this is working as intended.

**Comment by grasmanek94** [comment]
If this is intended then please do see this this report as 'never submitted' :)

**Comment by eidheim** [comment]
The `command does not exist` message also is output when using MSYS2 since it's using `sh -c`, and you can probably get a similar message on Windows outside MSYS2, but you would have to start the process with `cmd /C` (see https://github.com/eidheim/tiny-process-library/blob/master/examples.cpp#L120). `cmd` would then output this error message, but bear in mind that I did not create the `cmd` examples and have not tried this library outside MSYS2 on Windows. 

Sorry for the confusion, but the difference in Windows and Posix process creation is significant. It is manageable though when working within MSYS2 since you have access to the `sh` command and other commands typically found on a Posix system.
